### DISCUSSION TITLE
How to get the memory address of ValueType fields

### DISCUSSION THREAD

**Discussion by SamueleD78** [vfsfitvnm/frida-il2cpp-bridge#189] [Q&A]
As in the title, i need to recover the memory address of a field in a struct.
Let's say my code it's this one:

```
let tmp = Il2Cpp.alloc(MyStructType.valueSize);
const inst = new Il2Cpp.ValueType(tmp, MyStructType.type);
```
If i `console.log(tmp)`, i have the struct base address.
Now, let's suppose the struct has a field `index`. I usually recover the value this way:

```
console.log(inst.field<Il2Cpp.ValueType>("index").value);
```

But what if want to recover the address that field has in memory?
i tried with `console.log(inst.field<Il2Cpp.ValueType>("index").handle)` but of course it's wrong.
since it's the first element of the structure, the address should be the same as `tmp`.
Of course this example it's simple, i know i could just dump tmp. What i need it's to recover the address accessing from the `.field()` functions.

Is there a way?

**Comment by vfsfitvnm** [comment]
[Here](https://github.com/vfsfitvnm/frida-il2cpp-bridge/blob/master/src/il2cpp/structs/field.ts#L94-L97) I calculate the handle.

If the field is static, there isn't a clear way to do so.
